contents/blog/decouple-deployment-from-release.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Why (and how) you should decouple deployment from release
+date: 2023-09-07
+author: ["ian-vanagas"]
+showTitle: true
+rootpage: /blog
+sidebar: Blog
+hideAnchor: true
+featuredImage: ../images/blog/green-blog-image.jpg
+featuredImageType: full
+category: Engineering
+tags:
+ - Guides
+ - Explainers
+---
+
+You build software for others to use. This means you need a process for getting that software to those people. This is easier than ever thanks to the internet, hosting services, code management platforms, integration pipelines, and more innovations.
+
+What remains a challenge is delivering code to people **safely**. A best practice for doing this is decoupling deployment and release, but what does this even mean? This post goes over the differences between the two, why you should decouple them, examples, and how to get started.
+
+## What are the differences between deployment and release?
+
+The process of delivering code starts with writing and testing code locally. After completing a piece of code, it often goes through reviews and more tests before integrating with existing code. 
+
+Once integrated, the new code deploys. Deployment means getting code from a local machine or repository into production. Your application or build containing the new code becomes accessible on production infrastructure.
+
+After deploying successfully, the code releases for users to start using it. The application exposes in production and the functionality becomes available. A release comes after a deployment, and by default, couples to it. This means it happens immediately after a deployment succeeds.
+
+![Delivery process](../images/blog/decouple-deployment-from-release/delivery.png)
+
+This whole process is the software delivery process. Deployment and release are two separate, but critical parts of it.
+
+## Why should you decouple deployment and release?
+
+Without restrictions, deployed code is released code. This coupling of deployment and release can cause two main issues:
+
+1. Problems isolated to the new code due to deployment. For example, a value saving locally but not in production due to an unexpected interaction with the production database.
+
+2. Problems with the entire release due to new code. This is a broken release. For example, values created by new features corrupt other data unexpectedly in the production database.
```

Comment:
Took me a while to understand what "Problems isolated to the new code due to deployment" and "Problems with the entire release due to new code" mean. I dont think its super clear. Can we make it simpler, and with an example?

Maybe something like this:

>  This coupling of deployment and release can cause two main issues:
>
> 1. Their may be bugs in your new code, causing your feature not to work as expected
> 2. Their may be bugs in your code that cause your entire app to crash, causing a broken release

Reply:
Rewrote